virttest/cartesian_config.py

```diff
@@ -606,15 +606,15 @@ def _substitution(value, d):
     if "$" in value:
         start = 0
         st = ""
+        d_flat = _drop_suffixes(d)
         try:
             match = match_substitute.search(value, start)
             while match:
-                d_flat = _drop_suffixes(d)
-                val = eval(match.group(1), None, d_flat)
+                val = d_flat[match.group(1)]
                 st += value[start:match.start()] + str(val)
                 start = match.end()
                 match = match_substitute.search(value, start)
-        except Exception:
+        except KeyError:
```

Comment:
I assume investigation in the original commit where this code was introduced does not mention why the choise used to be an `Exception` instance. That is, I hope this change will not introduce some issues some issues the author was trying to contain. My testing did not reveal problem here so I guess it is reasonable.